components/process.rst

```diff
@@ -100,6 +100,33 @@ with a non-zero code)::
         echo $exception->getMessage();
     }
 
+.. tip::
+
+    .. versionadded:: 3.3
+        The ability to define commands as array of arguments was introduced in Symfony 3.3.
+
+    When defining commands as arrays, its arguments are escaped for you and you should not
+    escape them any further. You can also define commands as strings. When doing so, the
+    command-line is parsed by the underlying shell of your operating system. This allows
+    using e.g. stream redirections or conditional execution, but this is a lot less portable
+    since each OS (esp. Windows vs Unix-like) handles escaping and parsing differently.
+    When passing a command-line string, it becomes your responsibility to deal with this.
+
+    Using array of arguments is the recommended way to define commands.
+
+    If you need to define commands as strings, variable arguments should be passed as
+    environment variables using the second argument of the ``run()``, ``mustRun()`` or
+    ``start()`` methods. Referencing them in command-line strings is OS-dependent::
+
+        // On Unix-like OSes
+        $process = new Process('echo "$MESSAGE"');
+
+        // On Windows
+        $process = new Process('echo %MESSAGE%');
+
+        // On both Unix-like and Windows
+        $process->run(null, 'MESSAGE' => 'Something to output']);
```

Comment:
The command is missing or what did I miss?

Reply:
The command is `echo ...` This line is the `->run()` method call, which is the same for all OS. Only the command definition depends on the OS.

Reply:
I see, when I read this part it feels like it wants to show me the difference between using a string or an array value which is not the case. That's a bit confusing. 
Not sure you understand what I mean. So when I read `On both Unix-like and Windows` it sounds to me like the example should tell me how I could make the command run on both systems without dealing with escaping. Oh man, this is hard to explain 😄